draft-irtf-cfrg-opaque.md

```diff
@@ -1684,19 +1700,121 @@ protocols such as TLS.
+  different application uses of OPAQUE. This is a mechanism for the AKE
+  component and is best practice as for domain separation between different
+  applications of the protocol. This change was made to allow different
+  applications to use OPAQUE without risk of cross-protocol attacks.
+- Servers use a separate identifier for computing OPRF evaluations and
+  indexing into the password file storage, called the credential_identifier.
+  This allows clients to change their application-layer identity
+  (client_identity) without inducing server-side changes, e.g., by changing
+  an email address associated with a given account. This mechanism is part
+  of the derivation of OPRF keys via a single OPRF. As long as the derivation
+  of different OPRF keys from a single OPRF have different PRF inputs, the
+  protocol is secure. The choice of such inputs is up to the application.
+
+The following list enumerates notable differences and refinements from the original
+cryptographic design in {{OPAQUE}} and the corresponding CFRG document
+{{I-D.krawczyk-cfrg-opaque-03}} that were made to make this specification
+suitable for interoperable implementations.
+
+- {{OPAQUE}} used a generic prime-order group for the DH-OPRF and HMQV operations,
+  and includes necessary prime-order subgroup checks when receiving attacker-controlled
+  values over the wire. This specification instantiates the prime-order group using for
+  3DH using prime-order groups based on elliptic curves, as described in
+  {{I-D.irtf-cfrg-voprf, Section 2.1}}. This specification also delegates OPRF group
+  choice and operations to {{!I-D.irtf-cfrg-voprf}}. As such, the prime-order group as used
+  in the OPRF and 3DH as specified in this document both adhere to the requirements as
+  {{OPAQUE}}.
+- {{OPAQUE}} specified DH-OPRF (see Appendix B) to instantiate
+  the OPRF functionality in the protocol. A critical part of DH-OPRF is the
+  hash-to-group operation, which was left undefined in the original analysis.
+  However, the requirements for this operation were included. This specification
+  instantiates the OPRF functionality based on the {{I-D.irtf-cfrg-voprf}}, which
+  is identical to the DH-OPRF functionality in {{OPAQUE}} and, concretely, uses
+  the hash-to-curve functions in {{?I-D.irtf-cfrg-hash-to-curve}}. All hash-to-curve
+  methods in {{I-D.irtf-cfrg-hash-to-curve}} are compliant with the requirement
+  in {{OPAQUE}}, namely, that the output be a member of the prime-order group.
+- {{OPAQUE}} and {{I-D.krawczyk-cfrg-opaque-03}} both used HMQV as the AKE
+  for the protocol. However, this document fully specifies 3DH instead of HMQV
+  (though a sketch for how to instantiate OPAQUE using HMQV is included in {{hmqv-sketch}}).
+  Since 3DH satisfies the essential requirements for the AKE as described in {{OPAQUE}}
+  and {{I-D.krawczyk-cfrg-opaque-03}}, as described in {{security-analysis}}, this change
```

Comment:
change 'as described in {{security-analysis}}' with 'as recalled in {{security-analysis}}'